ktor-network/ktor-network-quic/jvmAndNix/src/io/ktor/network/quic/frames/FrameReader.kt

```diff
@@ -0,0 +1,450 @@
+/*
+ * Copyright 2014-2023 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.quic.frames
+
+import io.ktor.network.quic.bytes.*
+import io.ktor.network.quic.errors.*
+import io.ktor.network.quic.packets.*
+import io.ktor.network.quic.util.*
+import io.ktor.utils.io.core.*
+
+internal object FrameReader {
+    suspend inline fun readFrame(
```

Comment:
just want to mention, that making this method `inline` is dangerous, as it will increase amount of bytecode produced if it's will be called in a lot of places

Reply:
as I see it now, it should be called only in one-two places but many many times and I placed the `inline` modifier to inline the lambda argument

Reply:
I'll leave it as it is for now, and will change the approach if the future if it would be needed